src/community/contract_oracle.ts

```diff
@@ -0,0 +1,11 @@
+import { CommunityContractData } from './ardrive_contract_oracle';
+
+/** An oracle interface responsible for reading contracts and retrieving the ArDrive Community Contract */
+export interface ContractOracle extends ContractReader {
+	getCommunityContract(): Promise<CommunityContractData>;
+	getTipSettingFromContractSettings(): Promise<number>;
+}
+
+export interface ContractReader {
+	readContract(txId: string): Promise<unknown>;
```

Comment:
Nit: use alias types for `string` and `number` in the file to held readability. We can wait until we unify the price estimator code and this code in core.